packages/desktop-gui/cypress/integration/settings_spec.coffee

```diff
@@ -267,3 +269,21 @@ describe "Settings", ->
       @goToSettings()
 
       cy.contains("h5", "Record Keys").should("not.exist")
+
+  context "when configFile is false", ->
+    beforeEach ->
+      @openProject.resolve(_.assign({ configFile: false }, @config))
+      @goToSettings()
+      cy.contains("Configuration").click()
+
+    it "notes that cypress.json is disabled", ->
+      cy.contains("set from cypress.json (currently disabled by --config-file false)").should("exist")
+
+  context "when configFile is set", ->
+    beforeEach ->
+      @openProject.resolve(_.assign({ configFile: "special-cypress.json" }, @config))
+      @goToSettings()
+      cy.contains("Configuration").click()
+
+    it "notes that a custom config is in use", ->
+      cy.contains("set from custom config file special-cypress.json").should("exist")
```

Comment:
Can we add a test where the user sets the config file to an invalid file type (if JSON is required) - say they set it to a `special-cypress.ts` file, for example - what happens?

Reply:
If a user tries to use a file that doesn't contain JSON, the existing behavior still works (codeship.json has YAML in it):

![image](https://user-images.githubusercontent.com/1151760/52282164-c2cd1c80-292d-11e9-85fc-46d3f723a2ee.png)

The same error is thrown in run mode.

Reply:
Should there be a filename restriction (only `.json` files) as well?

Reply:
Yeah, what I was asking is, what happens if I pass `--configFile='foo.yml'`, not if I put invalid json data in a `.json` file?

Reply:
It would try to read and write JSON from foo.yml, there's no filename validation there

Reply:
Which is probably fine behavior, as it doesn't really matter what the extension is... they could be trying to use "--config-file cypress.json.test" to debug something, for example.